tests/vm_state_test.go

```diff
@@ -0,0 +1,232 @@
+		}
+
+		migrateVMI := func(vmi *v1.VirtualMachineInstance) {
+			By("Migrating the VMI")
+			checks.SkipIfMigrationIsNotPossible()
+			migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+			tests.RunMigrationAndExpectCompletion(virtClient, migration, tests.MigrationWaitTime)
+
+		}
+
+		addDataToTPM := func(vmi *v1.VirtualMachineInstance) {
+			By("Storing a secret into the TPM")
+			// https://www.intel.com/content/www/us/en/developer/articles/code-sample/protecting-secret-data-and-keys-using-intel-platform-trust-technology.html
+			// Not sealing against a set of PCRs, out of scope here, but should work with a carefully selected set (at least PCR1 was seen changing accross reboots)
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "tpm2_createprimary -Q --hierarchy=o --key-context=prim.ctx\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "echo MYSECRET | tpm2_create --hash-algorithm=sha256 --public=seal.pub --private=seal.priv --sealing-input=- --parent-context=prim.ctx\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "tpm2_load -Q --parent-context=prim.ctx --public=seal.pub --private=seal.priv --name=seal.name --key-context=seal.ctx\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "tpm2_evictcontrol --hierarchy=o --object-context=seal.ctx 0x81010002\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "tpm2_unseal -Q --object-context=0x81010002\n"},
+				&expect.BExp{R: "MYSECRET"},
+			}, 300)).To(Succeed(), "failed to store secret into the TPM")
+		}
+
+		checkTPM := func(vmi *v1.VirtualMachineInstance) {
+			By("Ensuring the TPM is still functional and its state carried over")
+			ExpectWithOffset(1, console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "tpm2_unseal -Q --object-context=0x81010002\n"},
+				&expect.BExp{R: "MYSECRET"},
+			}, 300)).To(Succeed(), "the state of the TPM did not persist")
+		}
+
+		addDataToEFI := func(vmi *v1.VirtualMachineInstance) {
+			By("Creating an efivar")
+			cmd := `printf "\x07\x00\x00\x00\x42" > /sys/firmware/efi/efivars/kvtest-12345678-1234-1234-1234-123456789abc`
+			err = console.RunCommand(vmi, cmd, 10*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		checkEFI := func(vmi *v1.VirtualMachineInstance) {
+			By("Ensuring the efivar is present")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "hexdump /sys/firmware/efi/efivars/kvtest-12345678-1234-1234-1234-123456789abc\n"},
+				&expect.BExp{R: "0042"},
+			}, 10)).To(Succeed(), "expected efivar is missing")
+		}
+
+		DescribeTable("[Serial]should persist VM state of", Serial, func(withTPM, withEFI bool, ops ...string) {
+			By("Setting the backend storage class to the default for RWX FS")
+			storageClass, exists := libstorage.GetRWXFileSystemStorageClass()
+			Expect(exists).To(BeTrue(), "No RWX FS storage class found")
```

Comment:
I missed this in my review, it's probably not fair to always expect a RWX storage class to exist, this might cause problems in test environments where it doesn't, can you add a Ginkgo label/decorator to wrap this?

Or does kubevirtci make sure that a RWX storage class always exist?

Reply:
> I missed this in my review, it's probably not fair to always expect a RWX storage class to exist, this might cause problems in test environments where it doesn't, can you add a Ginkgo label/decorator to wrap this?

The test is marked as sig-storage, which always has RWX FS storage. I don't see anything else on the other tests that use `GetRWXFileSystemStorageClass()`.

> 
> Or does kubevirtci make sure that a RWX storage class always exist?

Yes the sig-storage lanes do.